src/main/java/ch/njol/skript/expressions/ExprServerIcon.java

```diff
@@ -0,0 +1,129 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.ScriptLoader;
+import ch.njol.util.coll.CollectionUtils;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.event.server.PaperServerListPingEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.event.server.ServerListPingEvent;
+import org.bukkit.util.CachedServerIcon;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Server Icon")
+@Description({"Icon of the server in the server list. The first pattern doesn't return anything and will not work outside of a <a href='events.html#server_list_ping'>server list ping</a> event. " +
+		"But can be set to an icon that loaded using the <a href='effects.html#EffLoadServerIcon'>load server icon</a> effect, or can be reset to the default icon.",
+		"The second pattern returns the default server icon (server-icon.png) and cannot be changed."})
+@Examples({"on script load:",
+		"	set {server-icons::default} to the default server icon"})
+@Since("INSERT VERSION")
+public class ExprServerIcon extends SimpleExpression<CachedServerIcon> {
+
+	static {
+		Skript.registerExpression(ExprServerIcon.class, CachedServerIcon.class, ExpressionType.PROPERTY,
+				"[the] [(shown|sent)] [server] icon",
+				"[the] default [server] icon");
+	}
+
+	private boolean isServerPingEvent;
+	private int pattern;
+
+	@SuppressWarnings("null")
+	private Kleenean delay;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		pattern = matchedPattern;
+		boolean isPaperEvent = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent") && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class);
+		isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class) || isPaperEvent;
+		if (!isServerPingEvent && pattern == 0) {
+			Skript.error("The 'shown' server icon expression can't be used outside of a server list ping event");
+			return false;
+		}
+		delay = isDelayed;
+		return true;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public CachedServerIcon[] get(Event e) {
+		if (isServerPingEvent && pattern == 0)
+			return null;
```

Comment:
nothing stops you from using reflection to access the icon from the event, or storing it in a weak hashmap

Reply:
Oh there is already a method to get it from the event on PaperSpigot, i forgot to add support for it. But i don't think to deal with that to support Spigot, already not an useful thing.

Reply:
if you can't return false on init then it should work in both places. just use reflection via a cached Field if not on paper